### ISSUE TITLE
Frida Native Hook #spawn#resume

### ISSUE THREAD

**Issue by ChaplinLi** [frida/frida#404]
I want to use frida to hook a demo apk ,Here is my code:
```
import frida
import sys

package_name = "com.example.hello"


def get_messages_from_js(message, data):
	if message['type'] == 'send':
		print("[*] {0}".format(message['payload']))
	else:
		print(message)
 
native_hook_code = """
	Java.perform(function(){
		send("Running Script");
		
		Interceptor.attach(Module.findExportByName("libhello.so","Java_com_example_hello_MainActivity_hello"),{
			onEnter: function(args) {
				send("Inside libhellomethod");
			},
			onLeave: function(retval){
				
				send("retval type:"+ typeof retval);
				send("retval : " + retval);
				send("Leave libhellomethod");
				newretval = ptr("0x00");
				retval.replace(newretval);
				
			}
	
		})
		
	});

"""

java_hook_code = """
	send("Running Script");
	Java.perform(function(){
		MainActivity = Java.use("com.example.hello.MainActivity");
		MainActivity.hello.implementation = function(v){
			send("hack hello");
			return "hack hello";		
		}
	});
"""

device = frida.get_device_manager().enumerate_devices()[-1]
pid = device.spawn(["com.example.hello"])
session = device.attach(pid)
print("[*] Attach Application id:",pid)

device.resume(pid)
print("[*] Application onResume")

#script = session.create_script(java_hook_code)
script = session.create_script(native_hook_code)
script.on('message',get_messages_from_js)
script.load()
print("[*] Start Script")

sys.stdin.read()

```

Anyway Java hook is worked,but using native hook it will throw an exception:

```
[*] Attach Application id: 24727
[*] Application onResume
[*] Start Script
[*] Running Script
{'stack': 'Error: expected a pointer\n    at frida/runtime/core.js:471\n    at [anon] (script1.js:19)\n    at frida/node_modules/frida-java/lib/vm.js:39\n    at y (frida/node_modules/frida-java/index.js:325)\n    at frida/node_modules/frida-java/index.js:303\n    at input:1', 'fileName': 'frida/runtime/core.js', 'columnNumber': 1, 'description': 'Error: expected a pointer', 'lineNumber': 471, 'type': 'error'}
```
It seems like didn't find the libhello.so file.Why i didn't use the manual way to attach my application because it will throw anthor exception 
`try "sudo sysctl kernel.yama.ptrace_scope=0", or run Frida as root`

please tell me how to fixed,thanks.

**Comment by brian-leidos** [comment]
That means your findExportByName isn't working. Best bet is to use Frida interactive to get it right:
$ frida -U com.example.hello
[device::app]-> Module.findExportByName(....)

or you might try to use ApiResolver to get the names...
[device::app]-> r = new ApiResolver("module")
[device::app]-> r.enumerateMatches("exports:libhello.so!*", {onMatch: function(match) } console.log(match.name)}, onComplete: function() { } })

**Comment by ChaplinLi** [comment]
Yes,i have try to use Frida interactive to spawn it,and it can find method inside  "libhello.so" .I supspected whether if the Interceptor can't attach it
```
[device::app]-> Module.findExportByName("libhello.so","Java_com_example_hello_MainActivity_hello")
"0x7898af25"
```

**Comment by ChaplinLi** [comment]
Now i improve my code to display more infomation,it seems like frida can't find the function inside libhello.so.I'm thinking could i get this funtion's address,that i can attach it.
```
code:
var func = Module.findExportByName("libhello.so","Java_com_example_hello_MainActivity_hello");
		send("Find Function :"+func);

logcat:
[*] Attach Application id: 20556
[*] Application onResume
[*] Start Script
[*] Running Script
[*] Find Function :null
```
By the way,i can use this command "$ frida -U com.example.hello" to find the function address,but can't do the same thing on the script.

**Comment by oleavr** [comment]
Should be fixed in 10.6.40. Cheers!
